source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -0,0 +1,282 @@
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        ViInt32 get_int32_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViInt32Request request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViInt32Response response;
+            ::grpc::Status status = GetStub()->GetAttributeViInt32(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        ViReal64 get_real64_attribute(const char* channel_name, dmm::NiDMMAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViReal64Request request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViReal64Response response;
+            ::grpc::Status status = GetStub()->GetAttributeViReal64(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+    private:
+        std::shared_ptr<::grpc::Channel> channel_;
+        std::unique_ptr<::nidevice_grpc::Session> driver_session_;
+        std::unique_ptr<dmm::NiDMM::Stub> nidmm_stub_;
+        std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+        std::unique_ptr<dmm::NiDMMLibrary> nidmm_library_;
+        std::unique_ptr<dmm::NiDMMService> nidmm_service_;
+        std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiDMMDriverApiTest, NiDMMSelfTest_SendRequest_SelfTestCompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  dmm::SelfTestRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  dmm::SelfTestResponse response;
+
+  ::grpc::Status status = GetStub()->SelfTest(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+  EXPECT_EQ(0, response.self_test_result());
+  EXPECT_LT(0, strlen(response.self_test_message().c_str()));
+}
+
+TEST_F(NiDMMDriverApiTest, NiDMMReset_SendRequest_ResetCompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  dmm::ResetRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  dmm::ResetResponse response;
+
+  ::grpc::Status status = GetStub()->Reset(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiDMMDriverApiTest, NiDMMSetViReal64Attribute_SendRequest_GetViReal64AttributeMatches)
```

Comment:
Matches what? How about `SetViReal64Attribute_GetViReal64Attribute_ValueMatchesSetValue`

Reply:
changed it